### PR TITLE
[Intel GPU] DeepSeek V4 11/N: support fp8_paged_mqa_logits_triton from sgl-kernel to run on XPU

### PR DESCRIPTION
**Pull request by polisettyvarma**
Add unit test

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

---
### CI States

Latest PR Test (Base): :x: [Run #29313758639](https://github.com/sgl-project/sglang/actions/runs/29313758639)
Latest PR Test (Extra): :x: [Run #29313758524](https://github.com/sgl-project/sglang/actions/runs/29313758524)

### CODE REVIEW

File: python/sglang/srt/layers/attention/dsv4/indexer.py [hunk 1]
```diff
@@ -525,6 +525,10 @@ def forward_c4_indexer(
                 fn = fp8_paged_mqa_logits_torch_sm120
             else:
                 fn = fp8_paged_mqa_logits_torch
+        elif envs.SGLANG_FP8_PAGED_MQA_LOGITS_TRITON.get():
+            from .triton_fp8_paged_mqa_logits import fp8_paged_mqa_logits_triton
```
Comment (jianan-gu) [inline python/sglang/srt/layers/attention/dsv4/indexer.py]: For functionality, is there a reason that we have to add this triton implementation, rather than use the torch native based one which is already there? Since also for best perf, we anyway need sgl-kernel kernel based one.

Comment (polisettyvarma) [reply python/sglang/srt/layers/attention/dsv4/indexer.py]: torch implementation gives error/OOM, not necessarily we need sgl-kernel everywhere, triton itself gives good perf boost.

Comment (jianan-gu) [reply python/sglang/srt/layers/attention/dsv4/indexer.py]: Do you know the rootcase if the error/OOM issue met with torch impl?
Could you add more details (maybe in PR description) about such perf boost you observed and analyzed? Specific for this kernel, triton-based path is not added for cuda or aiter for best perf, it is not about sgl-kernel everywhere but about thinking how eventually we deliver best perf.

Comment (polisettyvarma) [reply python/sglang/srt/layers/attention/dsv4/indexer.py]: discussed offline

File: python/sglang/srt/layers/attention/dsv4/indexer.py [hunk 2]
```diff
@@ -668,6 +668,10 @@ def forward_c4_indexer(
                 fn = fp8_paged_mqa_logits_torch_sm120
             else:
                 fn = fp8_paged_mqa_logits_torch
+        elif is_xpu():
+            from sgl_kernel import fp8_paged_mqa_logits_triton
```
Comment (jianan-gu) [inline python/sglang/srt/layers/attention/dsv4/indexer.py:672]: can we rename with just `fp8_paged_mqa_logits`?

Comment (polisettyvarma) [reply python/sglang/srt/layers/attention/dsv4/indexer.py:672]: i can't use due to existing kernel where we get OOM

Comment (polisettyvarma) [reply python/sglang/srt/layers/attention/dsv4/indexer.py:672]: added TODO as discussed offline

File: python/sglang/srt/layers/attention/dsv4/indexer.py [hunk 3]
```diff
@@ -668,6 +668,12 @@ def forward_c4_indexer(
...
+
+            # TODO: switch from triton to SYCL when OOM is resolved
```
Comment (jianan-gu) [inline python/sglang/srt/layers/attention/dsv4/indexer.py:674]: Please do one more check on latest sgl-kernel-xpu repo to see if the issue is still valid.

Comment (polisettyvarma) [reply python/sglang/srt/layers/attention/dsv4/indexer.py:674]: issue is still there

**Review by jianan-gu — APPROVED** [review]

**Review by mingfeima — APPROVED** [review]
